src/funding/funding.test.js

```diff
@@ -0,0 +1,106 @@
+/* @flow */
+import { COMPONENTS, FUNDING } from "@paypal/sdk-constants/src";
+import { describe, expect } from "vitest";
+
+import { isFundingEligible } from "./funding";
+
+const defaultMockFundingOptions = {
+  platform: "desktop",
+  fundingEligibility: {
+    paylater: {
+      eligible: true,
+      vaultable: false,
+    },
+    venmo: {
+      eligible: true,
+      vaultable: true,
+    },
+    sepa: {
+      eligible: false,
+    },
+    card: {
+      eligible: true,
+      vendors: {
+        visa: {
+          eligible: true,
+          vaultable: false,
+        },
+        mastercard: {
+          eligible: true,
+          vaultable: false,
+        },
+        amex: {
+          eligible: true,
+          vaultable: false,
+        },
+      },
+    },
+  },
+};
+
+describe("Funding eligibility", () => {
+  test("should not be eligible if funding source is missing from fundingEligibility", () => {
+    const fundingEligible = isFundingEligible("fake_funding", {
+      fundingEligibility: { venmo: { eligible: true } },
+    });
+
+    expect(fundingEligible).toBe(false);
+  });
+
+  test("should not be eligible if displayOnly includes 'vaultable' and vaultable is false", () => {
+    const options = {
+      displayOnly: ["vaultable"],
+      ...defaultMockFundingOptions,
+    };
+    const fundingEligible = isFundingEligible(FUNDING.PAYLATER, options);
+
+    expect(fundingEligible).toBe(false);
+  });
+
+  test("card should not be eligible if displayOnly includes 'vaultable' and no vendors are vaultable", () => {
+    const options = {
+      displayOnly: ["vaultable"],
+      components: COMPONENTS.BUTTONS,
+      ...defaultMockFundingOptions,
+    };
+    const fundingEligible = isFundingEligible(FUNDING.CARD, options);
+
+    expect(fundingEligible).toBe(false);
+  });
+
+  test("card should be eligible if displayOnly includes 'vaultable' and any vendor is vaultable", () => {
```

Comment:
may want to add one more test case
```
amex: {
  eligible: false,
  vaultable: true,
},
```

Reply:
Thanks man! Is this possible though? From my understanding, it can't be vaultable if its not eligible, right?

Reply:
Thanks for taking time to review this! I also have to fix some type stuff that's throwing errors with `npm run typecheck`

Reply:
idk you wrote code that handles this lol: https://github.com/paypal/paypal-checkout-components/pull/2252/files#diff-f677f43c72524dd42a4f608dd81477d6a6d9e896205422f6c4aaa5847e9ffeafR49-R52

if you don't think that's possible, i would just remove the `vendor.eligible` check from your `.some` function

Reply:
Hahah ohhh.. you're right.  I just looked back at a sample query & card vendors can be vaultable but not eligible 😅 I'll add the test case. 😄